File: Modules/SEDO/EMS/SIE.Web.EMS/Equipments/Accounts/Commands/SelAccountMaintainCommand.js
SIE.defineCommand('SIE.Web.EMS.Equipments.Accounts.Commands.SelAccountMaintainCommand', {
    extend: 'SIE.Web.EMS.Common.Commands.SelMaintainProjectCommand',
    meta: { text: "选择", group: "edit", iconCls: "icon-PlaylistCheck icon-blue" },  
/**
     * canExecute 是否执行
     * @param {} view 当前视图
     * @returns {}
     */
    canExecute: function (view) {
        var datas = view.getData();
        if (datas && datas.isDirty())
            return false;

        var entity = view.getParent().getCurrent();
        if (entity != null && entity.data) {
            return entity.data.CreateBy !== null;
        }
        return false;
    }, 
});

File: Modules/SEDO/EMS/SIE.Web.EMS/Equipments/Accounts/Commands/SelAccountMaintainCommand.cs
using SIE.EMS.Common.Entity;
using SIE.EMS.Equipments;
using SIE.Web.Command;
using System.Collections.Generic;

namespace SIE.Web.EMS.Equipments.Accounts.Commands
{
    /// <summary>
    /// 添加保养项目
    /// </summary>
    [JsCommand("SIE.Web.EMS.Equipments.Accounts.Commands.SelAccountMaintainCommand")]
    public class SelAccountMaintainCommand : ViewCommand
    {
        /// <summary>
        /// 执行添加
        /// </summary>
        /// <param name="args">args</param>
        /// <param name="scope">scope</param>
        /// <returns>执行结果</returns>
        protected override object Excute(ViewArgs args, string scope)
        {
            var checkProjectInfos = args.Data.ToJsonObject<List<CheckProjectInfo>>();
            Check.NotNullOrEmpty(checkProjectInfos, nameof(checkProjectInfos));
            RT.Service.Resolve<EquipController>().SaveAccountMaintainCommand(checkProjectInfos);
            return true;
        }
    }
}
